File: yh_star/scripts/yh_star_sub.py
#!/usr/bin/python3
# -*- coding: utf8 -*-

import imp
import rospy
from yh_star.msg import yh_star_msg

def msgCallback(msg):

    rospy.loginfo("number is = %d", msg.data)

    num = msg.data
    star = ""

    for i in range(1, num//2 + 1):
        for j in range(i):
            star += "*"
        star += "\n"

    for i in range(num//2 + 1, num + 1):
        for j in range(num - i + 1):
            star += "*"
        star += "\n"

    print(star)


def listener():
    rospy.init_node("yh_star_sub", anonymous=True)
    rospy.Subscriber("yh_star_topic", yh_star_msg, msgCallback)

    rospy.spin()

if __name__ == "__main__":
    listener()

File: yh_star/src/yh_star_sub.cpp
#include "ros/ros.h" 
#include "yh_star/yh_star_msg.h"  
                                   

void msgCallback(const yh_star::yh_star_msg::ConstPtr& msg)   
{
    ROS_INFO("Number is = %d", msg->data);

    int num = msg->data;
    
    for (int i = 0; i < num/2; i++)
    {
        for (int j = 0; j < i+1; j++) printf("*");
        printf("\n");
    }
    for (int i = num/2; i < num ; i++)
    {
        for (int j = num; j > i; j--) printf("*");
        printf("\n");
    }


}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "yh_star_sub"); 
    ros::NodeHandle nh; 

    ros::Subscriber sub = nh.subscribe("yh_star_topic", 100, msgCallback);   
                                                                          

    ros::spin();   
                    

    return 0;

}